content/en/docs/ops/common-problems/security-issues/index.md

```diff
@@ -408,3 +408,84 @@ second ensure that [keys and certificates are being delivered](#keys-and-certifi
 
 If everything appears to be working so far, the next step is to verify that the right [authentication policy](/docs/tasks/security/authentication/authn-policy/)
 is applied and the right destination rules are in place.
+
+### Auto mTLS
+
+If you suspect [auto mTLS](/docs/reference/glossary#auto-mtls) is not working
+as expected, please first read the
+[documentation](/docs/tasks/security/authentication/authn-policy/#auto-mutual-tls)
+and understand that [workload level](/docs/tasks/security/authentication/authn-policy/#enable-mutual-tls-per-workload) 
+peer authentication policies require corresponding destination rule configuration.
+
+If you still think there's an issue, follow the instructions below for
+investigation.
+
+1. Get the client envoy cluster configuration to ensure auto mTLS is configured.
+You should see `transportSocketMatches` configured for the given Envoy cluster.
+
+    {{< text bash >}}
+    $ POD=$(kubectl get pod -l app=sleep -n foo -o jsonpath={.items..metadata.name})
+    $ istioctl proxy-config clusters ${POD}.foo -ojson
+    "transportSocketMatches": [
+            {
+                "name": "tlsMode-istio",
+                "match": {
+                    "tlsMode": "istio"
+                },
+                "transportSocket": {
+                    "name": "envoy.transport_sockets.tls",
+                    "typedConfig": {
+                        "@type": "type.googleapis.com/envoy.extensions.transport_sockets.tls.v3.UpstreamTlsContext",
+                        "commonTlsContext": ...
+            }
+            {
+                "name": "tlsMode-disabled",
+                "match": {},
+                "transportSocket": {
+                    "name": "envoy.transport_sockets.raw_buffer"
+                }
+            }
+    {{< /text >}}
+
+1. Check the destination pod to see if the destination endpoint has the label
+`tlsMode` equals to `istio`.
+
+    {{< text bash >}}
+    $ POD=$(kubectl get pod -l app=sleep -n foo -o jsonpath={.items..metadata.name})
+    $ ISTIOD=$(kubectl get pod -l app=istiod -o jsonpath={.items..metadata.name}))
+    $ kubectl exec -it ${ISTIOD} -nistio-system -- curl "localhost:8080/debug/edsz?proxyID=${POD}"
```

Comment:
Similar here, probably should include a `grep` command to quickly quickly find the relevant lines

Reply:
I would use `pilot-agent request GET config_dump?include_eds=true` on the pod, not istiod.

Reply:
done. i remember eds dump was introduced newish, but should  be fine.